Add DataStax as Cassandra test environment

As far as I confirmed, `5.1.5` works without any changes, but `6.8.18` fails to start a query runner due to below messages:
```bash
2022-01-01T14:37:44.296+0900	ERROR	main	Bootstrap	Uncaught exception in thread main
java.lang.RuntimeException: Schema tpch not found
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:122)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:505)
	at io.trino.testing.QueryAssertions.copyTable(QueryAssertions.java:339)
	at io.trino.testing.QueryAssertions.copyTable(QueryAssertions.java:331)
	at io.trino.testing.QueryAssertions.copyTpchTables(QueryAssertions.java:323)
	at io.trino.plugin.cassandra.DataStaxQueryRunner.createScyllaQueryRunner(DataStaxQueryRunner.java:61)
	at io.trino.plugin.cassandra.DataStaxQueryRunner.main(DataStaxQueryRunner.java:86)
	Suppressed: java.lang.Exception: SQL: CREATE TABLE IF NOT EXISTS customer AS SELECT * FROM tpch.tiny.customer
		at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:508)
		... 5 more
Caused by: io.trino.spi.connector.SchemaNotFoundException: Schema tpch not found
	at io.trino.plugin.cassandra.CassandraSession.getKeyspaceByCaseInsensitiveName(CassandraSession.java:274)
	at io.trino.plugin.cassandra.CassandraSession.getCaseSensitiveSchemaName(CassandraSession.java:154)
	at io.trino.plugin.cassandra.CassandraMetadata.createTable(CassandraMetadata.java:294)
	at io.trino.plugin.cassandra.CassandraMetadata.beginCreateTable(CassandraMetadata.java:278)
	at io.trino.spi.connector.ConnectorMetadata.beginCreateTable(ConnectorMetadata.java:528)
	at io.trino.metadata.MetadataManager.beginCreateTable(MetadataManager.java:955)
	at io.trino.sql.planner.optimizations.BeginTableWrite$Rewriter.createWriterTarget(BeginTableWrite.java:253)
	at io.trino.sql.planner.optimizations.BeginTableWrite$Rewriter.visitTableFinish(BeginTableWrite.java:197)
	at io.trino.sql.planner.optimizations.BeginTableWrite$Rewriter.visitTableFinish(BeginTableWrite.java:100)
	at io.trino.sql.planner.plan.TableFinishNode.accept(TableFinishNode.java:106)
	at io.trino.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:84)
	at io.trino.sql.planner.plan.SimplePlanRewriter$RewriteContext.lambda$defaultRewrite$0(SimplePlanRewriter.java:73)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Collections$2.tryAdvance(Collections.java:4747)
	at java.base/java.util.Collections$2.forEachRemaining(Collections.java:4755)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at io.trino.sql.planner.plan.SimplePlanRewriter$RewriteContext.defaultRewrite(SimplePlanRewriter.java:74)
	at io.trino.sql.planner.plan.SimplePlanRewriter.visitPlan(SimplePlanRewriter.java:38)
	at io.trino.sql.planner.plan.SimplePlanRewriter.visitPlan(SimplePlanRewriter.java:22)
	at io.trino.sql.planner.plan.PlanVisitor.visitOutput(PlanVisitor.java:49)
	at io.trino.sql.planner.plan.OutputNode.accept(OutputNode.java:83)
	at io.trino.sql.planner.plan.SimplePlanRewriter.rewriteWith(SimplePlanRewriter.java:32)
	at io.trino.sql.planner.optimizations.BeginTableWrite.optimize(BeginTableWrite.java:85)
	at io.trino.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:230)
	at io.trino.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:215)
	at io.trino.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:210)
	at io.trino.execution.SqlQueryExecution.doPlanQuery(SqlQueryExecution.java:466)
	at io.trino.execution.SqlQueryExecution.planQuery(SqlQueryExecution.java:447)
	at io.trino.execution.SqlQueryExecution.start(SqlQueryExecution.java:388)
	at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:243)
	at io.trino.dispatcher.LocalDispatchQuery.lambda$startExecution$7(LocalDispatchQuery.java:143)
	at io.trino.$gen.Trino_testversion____20220101_053741_3.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
	Suppressed: java.lang.Exception: Current plan:
                Output[rows]
                │   Layout: [rows:bigint]
                └─ TableCommit[cassandra.tpch.customer]
                   │   Layout: [rows:bigint]
                   └─ LocalExchange[SINGLE] ()
                      │   Layout: [partialrows:bigint, fragment:varbinary]
                      └─ RemoteExchange[GATHER]
                         │   Layout: [partialrows:bigint, fragment:varbinary]
                         └─ TableWriter
                            │   Layout: [partialrows:bigint, fragment:varbinary]
                            │   custkey := custkey
                            │   name := name
                            │   address := address
                            │   nationkey := nationkey
                            │   phone := phone
                            │   acctbal := acctbal
                            │   mktsegment := mktsegment
                            │   comment := comment
                            └─ LocalExchange[SINGLE] ()
                               │   Layout: [custkey:bigint, name:varchar(25), address:varchar(40), nationkey:bigint, phone:varchar(15), acctbal:double, mktsegment:varchar(10), comment:varchar(117)]
                               └─ RemoteExchange[REPARTITION]
                                  │   Layout: [custkey:bigint, name:varchar(25), address:varchar(40), nationkey:bigint, phone:varchar(15), acctbal:double, mktsegment:varchar(10), comment:varchar(117)]
                                  └─ TableScan[tpch:tiny:customer]
                                         Layout: [custkey:bigint, name:varchar(25), address:varchar(40), nationkey:bigint, phone:varchar(15), acctbal:double, mktsegment:varchar(10), comment:varchar(117)]
                                         nationkey := tpch:nationkey
                                         mktsegment := tpch:mktsegment
                                         address := tpch:address
                                         phone := tpch:phone
                                         custkey := tpch:custkey
                                         name := tpch:name
                                         comment := tpch:comment
                                         acctbal := tpch:acctbal

		at io.trino.sql.planner.optimizations.BeginTableWrite.optimize(BeginTableWrite.java:91)
		... 12 more
```

https://hub.docker.com/r/datastax/dse-server

--- a/plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java
+++ b/plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java
@@ -20,6 +20,7 @@ import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
 import com.datastax.oss.driver.api.core.config.ProgrammaticDriverConfigLoaderBuilder;
 import com.datastax.oss.driver.api.core.cql.ResultSet;
 import com.datastax.oss.driver.api.core.cql.Row;
+import com.google.common.collect.ImmutableMap;
 import com.google.common.io.Resources;
 import io.airlift.json.JsonCodec;
 import io.airlift.log.Logger;
@@ -32,6 +33,7 @@ import java.io.IOException;
 import java.net.InetSocketAddress;
 import java.nio.file.Path;
 import java.util.List;
+import java.util.Map;
 import java.util.concurrent.TimeoutException;
 
 import static com.datastax.oss.driver.api.core.config.DefaultDriverOption.CONTROL_CONNECTION_AGREEMENT_TIMEOUT;
@@ -73,15 +75,23 @@ public class CassandraServer
     public CassandraServer(String imageName)
             throws Exception
     {
+        this(imageName, ImmutableMap.of(), "/etc/cassandra/cassandra.yaml", "cu-cassandra.yaml");
+    }
+
+    public CassandraServer(String imageName, Map<String, String> environmentVariables, String configPath, String configFileName)
+            throws Exception
+    {
         log.info("Starting cassandra...");
 
         this.dockerContainer = new GenericContainer<>(imageName)
                 .withExposedPorts(PORT)
-                .withCopyFileToContainer(forHostPath(prepareCassandraYaml()), "/etc/cassandra/cassandra.yaml");
+                .withCopyFileToContainer(forHostPath(prepareCassandraYaml(configFileName)), configPath)
+                .withEnv(environmentVariables)
+                .withStartupTimeout(java.time.Duration.ofMinutes(10));
         this.dockerContainer.start();
 
         ProgrammaticDriverConfigLoaderBuilder driverConfigLoaderBuilder = DriverConfigLoader.programmaticBuilder();
-        driverConfigLoaderBuilder.withDuration(REQUEST_TIMEOUT, java.time.Duration.ofSeconds(12));
+        driverConfigLoaderBuilder.withDuration(REQUEST_TIMEOUT, java.time.Duration.ofSeconds(30));
         driverConfigLoaderBuilder.withString(PROTOCOL_VERSION, ProtocolVersion.V3.name());
         driverConfigLoaderBuilder.withDuration(CONTROL_CONNECTION_AGREEMENT_TIMEOUT, java.time.Duration.ofSeconds(30));
         // allow the retrieval of metadata for the system keyspaces
@@ -111,10 +121,10 @@ public class CassandraServer
         this.session = session;
     }
 
-    private static String prepareCassandraYaml()
+    private static String prepareCassandraYaml(String fileName)
             throws IOException
     {
-        String original = Resources.toString(getResource("cu-cassandra.yaml"), UTF_8);
+        String original = Resources.toString(getResource(fileName), UTF_8);
 
         Path tmpDirPath = createTempDirectory(null);
         Path dataDir = tmpDirPath.resolve("data");
@@ -122,7 +132,7 @@ public class CassandraServer
 
         String modified = original.replaceAll("\\$\\{data_directory\\}", dataDir.toAbsolutePath().toString());
 
-        File yamlFile = tmpDirPath.resolve("cu-cassandra.yaml").toFile();
+        File yamlFile = tmpDirPath.resolve(fileName).toFile();
         yamlFile.deleteOnExit();
         write(modified, yamlFile, UTF_8);
 

--- a/plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java
+++ b/plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java
@@ -67,15 +67,15 @@ public class CassandraServer
     public CassandraServer()
             throws Exception
     {
-        this("2.2");
+        this("cassandra:2.2");
     }
 
-    public CassandraServer(String cassandraVersion)
+    public CassandraServer(String imageName)
             throws Exception
     {
         log.info("Starting cassandra...");
 
-        this.dockerContainer = new GenericContainer<>("cassandra:" + cassandraVersion)
+        this.dockerContainer = new GenericContainer<>(imageName)
                 .withExposedPorts(PORT)
                 .withCopyFileToContainer(forHostPath(prepareCassandraYaml()), "/etc/cassandra/cassandra.yaml");
         this.dockerContainer.start();

--- a/plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraLatestConnectorSmokeTest.java
+++ b/plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraLatestConnectorSmokeTest.java
@@ -28,7 +28,7 @@ public class TestCassandraLatestConnectorSmokeTest
     protected QueryRunner createQueryRunner()
             throws Exception
     {
-        CassandraServer server = closeAfterClass(new CassandraServer("3.11.10"));
+        CassandraServer server = closeAfterClass(new CassandraServer("cassandra:3.11.10"));
         CassandraSession session = server.getSession();
         createTestTables(session, KEYSPACE, Timestamp.from(TIMESTAMP_VALUE.toInstant()));
         return createCassandraQueryRunner(server, ImmutableMap.of(), ImmutableMap.of(), REQUIRED_TPCH_TABLES);

--- a/plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestDatastaxConnectorSmokeTest.java
+++ b/plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestDatastaxConnectorSmokeTest.java
@@ -1,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.cassandra;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.QueryRunner;
+
+import java.sql.Timestamp;
+import java.util.Map;
+
+import static io.trino.plugin.cassandra.CassandraQueryRunner.createCassandraQueryRunner;
+import static io.trino.plugin.cassandra.CassandraTestingUtils.createTestTables;
+
+public class TestDatastaxConnectorSmokeTest
+        extends BaseCassandraConnectorSmokeTest
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        CassandraServer server = closeAfterClass(new CassandraServer(
+                "datastax/dse-server:6.8.25",
+                Map.of(
+                        "DS_LICENSE", "accept",
+                        "DC", "datacenter1"),
+                "/config/cassandra.yaml",
+                "cassandra-dse.yaml"));
+        CassandraSession session = server.getSession();
+        createTestTables(session, KEYSPACE, Timestamp.from(TIMESTAMP_VALUE.toInstant()));
+        return createCassandraQueryRunner(server, ImmutableMap.of(), ImmutableMap.of(), REQUIRED_TPCH_TABLES);
+    }
+}

--- a/plugin/trino-cassandra/src/test/resources/cassandra-dse.yaml
+++ b/plugin/trino-cassandra/src/test/resources/cassandra-dse.yaml
@@ -1,0 +1,20 @@
+endpoint_snitch: SimpleSnitch
+
+commitlog_sync: periodic
+commitlog_sync_period_in_ms: 10000
+
+partitioner: org.apache.cassandra.dht.Murmur3Partitioner
+
+seed_provider:
+    - class_name: org.apache.cassandra.locator.SimpleSeedProvider
+      parameters:
+          - seeds: "127.0.0.1"
+
+native_transport_port: 9142
+
+read_request_timeout_in_ms: 30000
+range_request_timeout_in_ms: 30000
+write_request_timeout_in_ms: 30000
+cas_contention_timeout_in_ms: 30000
+truncate_request_timeout_in_ms: 60000
+request_timeout_in_ms: 30000

